vkcs/compute/data_source_flavor.go

```diff
@@ -220,13 +299,33 @@ func dataSourceComputeFlavorRead(ctx context.Context, d *schema.ResourceData, me
 			"Please change your search criteria and try again.")
 	}
 
+	// if we find many flavors and the user sets the min_ram or min_disk values
+	// we give him the flavor with the minimum amount of RAM from the found flavors
+	if len(allFlavors) > 1 && (requiredFlavor.HasMinRAM || requiredFlavor.HasMinDisk) {
+		resIdx := 0
+		for idx, flavor := range allFlavors {
+			if flavor.RAM == allFlavors[resIdx].RAM && flavor.Disk < allFlavors[resIdx].Disk {
+				resIdx = idx
+			} else if flavor.RAM < allFlavors[resIdx].RAM {
+				resIdx = idx
+			}
+		}
+
+		return diag.FromErr(dataSourceComputeFlavorAttributes(d, computeClient, allFlavors[resIdx].ToFlavor()))
+	}
+
 	if len(allFlavors) > 1 {
 		log.Printf("[DEBUG] Multiple results found: %#v", allFlavors)
-		return diag.Errorf("Your query returned more than one result. " +
-			"Please try a more specific search criteria")
+		if len(allFlavors) > 2 {
+			return diag.Errorf("Found %d available flavors. Please try a more specific search criteria", len(allFlavors))
+		}
+		bytes, _ := json.MarshalIndent(allFlavors, "", "\t")
+
+		return diag.Errorf("Found %d available flavors. Available flavors:\n%s\n"+
+			"Please try a more specific search criteria", len(allFlavors), bytes)
```

Comment:
I see no reason for such descriptive diagnostics here, the older one is common among all data sources. They can enable logging if they want to see which flavors were selected while reading data source, which, in my opinion, is a rare case because a single object data source isn't a suitable tool for this.

Reply:
Remove print of available flavors, should i remove also it counts?

Reply:
Yup, I don't think this info helps somehow.